envoy/auth_server/config/config.go

```diff
@@ -0,0 +1,71 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/buildwithgrove/path/envoy/auth_server/auth"
+)
+
+const (
+	defaultEndpointIDExtractorType = auth.EndpointIDExtractorTypeURLPath
+	defaultPort                    = 10003
+)
+
+type GatewayConfig struct {
```

Comment:
1. Please document structurs
2. Please document the fields, when appropriate
3. Confirming that the sturcts AND fields should all be publically exposed?